### PR TITLE
added filter of weight

### PR DESCRIPTION
**Pull request by mjekson**

### CODE REVIEW

File: octodns_edgecenter/__init__.py [hunk 1]
```diff
@@ -167,11 +182,21 @@ def _build_pools(self, record, default_pool_name, value_transform_fn):
             value = {"value": value_transform_fn(rr["content"][0])}
             countries = meta.get("countries", []) or []
             continents = meta.get("continents", []) or []
+            # weight = meta.get("weight", []) or []
```
Comment (ross) [inline octodns_edgecenter/__init__.py]:
```suggestion
```

File: octodns_edgecenter/__init__.py [hunk 2]
```diff
@@ -167,11 +182,21 @@ def _build_pools(self, record, default_pool_name, value_transform_fn):
...
 
             if meta.get("default", False):
                 pools[default_pool_name]["values"].append(value)
                 defaults.append(value["value"])
                 continue
+            elif meta.get("weight", 0) > 0:
+                # elif len(weight) > 0:
```
Comment (ross) [inline octodns_edgecenter/__init__.py]:
```suggestion
```

File: octodns_edgecenter/__init__.py [hunk 3]
```diff
@@ -167,11 +182,21 @@ def _build_pools(self, record, default_pool_name, value_transform_fn):
...
+                value_weight = {
+                    "value": value_transform_fn(rr["content"][0]),
+                    "weight": meta.get("weight", 0),
```
Comment (ross) [inline octodns_edgecenter/__init__.py]: Fwiw you shouldn't need a default here since you're in an `elif` that checked for a non-default value `> 0`,

```suggestion
                    "weight": meta["weight"],
```

File: octodns_edgecenter/__init__.py [hunk 4]
```diff
@@ -261,6 +286,8 @@ def _data_for_CNAME(self, _type, record):
             "value": self._add_dot_if_need(defaults[0]),
         }
 
+    # _data_for_ALIAS = _data_for_CNAME
+
```
Comment (ross) [inline octodns_edgecenter/__init__.py]:
```suggestion
```

File: octodns_edgecenter/__init__.py [hunk 5]
```diff
@@ -387,15 +414,28 @@ def _should_ignore(self, record):
...
+        for fls in filters:
+            if fls.get("type", "") == "weighted_shuffle":
+                want_filters = 2
+                break
+        # resource_records = record.get("resource_records", [])
+        # for rr in record["resource_records"]:
+        #     meta = rr.get("meta", {}) or {}
+        #     if meta.get("weight", 0) > 0:
+        #         want_filters = 2
+        #         break
```
Comment (ross) [inline octodns_edgecenter/__init__.py]:
```suggestion
```

File: octodns_edgecenter/__init__.py [hunk 6]
```diff
@@ -435,25 +480,32 @@ def _params_for_dymanic(self, record):
                         meta.setdefault("countries", []).append(country)
                     else:
                         meta.setdefault("continents", []).append(continent)
-            else:
+            elif not weight:
                 meta["default"] = True
 
+            # if len(rule.data.get("weight", [])) > 0:
+            #     weight = True
+
```
Comment (ross) [inline octodns_edgecenter/__init__.py]:
```suggestion
```

File: tests/test_octodns_provider_edgecenter.py [hunk 7]
```diff
@@ -631,3 +631,211 @@ def test_apply(self):
     def test_provider_hierarchy(self):
         provider = EdgeCenterProvider("test_id", token="token")
         self.assertIsInstance(provider, _BaseProvider)
+
+
+class TestEdgeCenterProviderWeighted(TestCase):
+    # expected = Zone("unit.tests.", [])
```
Comment (ross) [inline tests/test_octodns_provider_edgecenter.py]:
```suggestion
```

**Review by ross — COMMENTED** [review]
Looks fine, most of the inline suggestions are just removing commented out code. There's one minor nit suggestion inline. Otherwise once those are sorted it should be ready to :ship:

**Comment by mjekson** [comment]
Yep:) fix it and  change to v0.0.2

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,6 @@
+## v0.0.2 - 2023-02-14 - Support the filter of weight
```
Comment (ross) [inline CHANGELOG.md]: Releases are done in standalone PRs with the one PR is one change mindset.

```suggestion
## v0.0.2 - 2023-02-?? - Support the filter of weight
```

File: octodns_edgecenter/__init__.py [hunk 9]
```diff
@@ -13,7 +13,7 @@
 from octodns.provider.base import BaseProvider
 from octodns.record import GeoCodes, Record
 
-__VERSION__ = '0.0.1'
+__VERSION__ = '0.0.2'
```
Comment (ross) [inline octodns_edgecenter/__init__.py]:
```suggestion
__VERSION__ = '0.0.1'
```

**Review by ross — APPROVED** [review]
Releases are done in standalone PRs. Otherwise this looks good.
